apps/src/StudioApp.js

```diff
@@ -670,6 +671,12 @@ StudioApp.prototype.handleClearPuzzle = function (config) {
     this.editor.setValue(resetValue);
 
     annotationList.clearRuntimeAnnotations();
+  } else if (this.scratch) {
+    const workspace = Blockly.getMainWorkspace();
+    workspace.clear();
+
+    const dom = Blockly.Xml.textToDom(config.level.startBlocks);
+    Blockly.Xml.domToWorkspace(dom, workspace);
```

Comment:
So this would mean we're using two different forks of Blockly? Our own for most stuff, and then this Scratch specific one for Scratch stuff?